core/primitives/src/epoch_manager.rs

```diff
@@ -139,6 +149,20 @@ impl AllEpochConfig {
             config.chunk_producer_kickout_threshold = 80;
             config.validator_selection_config.num_chunk_only_producer_seats = 200;
         }
+
+        // Adjust the number of block and chunk producers for all chains except
+        // mainnet, to make it easier to test the change.
+        if chain_id != crate::chains::MAINNET
+            && checked_feature!("stable", TestnetFewerBlockProducers, protocol_version)
+        {
+            let num_shards = config.shard_layout.num_shards() as usize;
+            // Decrease the number of block producers from 100 to 20.
+            config.num_block_producer_seats = 20;
+            config.num_block_producer_seats_per_shard =
+                vec![config.num_block_producer_seats; num_shards];
+            // Decrease the number of chunk producers.
+            config.validator_selection_config.num_chunk_only_producer_seats = 100;
+        }
```

Comment:
How will it affect Calimero? cc @chefsale @miraclx

Reply:
I would consider adding one more check before overriding genesis config: the date of creation of the network. Thus, newly created networks would respect the value from the genesis config.